--- a/cloudinit/config/cc_ssh.py	2016-12-23 16:37:45.000000000 +0000
+++ b/cloudinit/config/cc_ssh.py	2017-05-17 11:54:24.928000000 +0000
@@ -117,7 +117,7 @@
         {"%s_public" % k: (KEY_FILE_TPL % k + ".pub", 0o600)})
     PRIV_TO_PUB["%s_private" % k] = "%s_public" % k
 
-KEY_GEN_TPL = 'o=$(ssh-keygen -yf "%s") && echo "$o" root@localhost > "%s"'
+KEY_GEN_TPL = 'o=$(dropbearkey -y -f "%s") && echo "$o" root@localhost > "%s"'
 
 
 def handle(_name, cfg, cloud, log, _args):
@@ -164,7 +164,7 @@
             if os.path.exists(keyfile):
                 continue
             util.ensure_dir(os.path.dirname(keyfile))
-            cmd = ['ssh-keygen', '-t', keytype, '-N', '', '-f', keyfile]
+            cmd = ['dropbearkey', '-t', keytype, '-f', keyfile]
 
             # TODO(harlowja): Is this guard needed?
             with util.SeLinuxGuard("/etc/ssh", recursive=True):
